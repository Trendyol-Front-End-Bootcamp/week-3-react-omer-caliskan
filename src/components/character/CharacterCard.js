import React from 'react';
import '../../assets/css/characterCard.css';

function CharacterCard({char}) {
    return (
        <div className="list-item">
            <h3>{char.name}</h3>
            <img src={char.image} alt="Rick and Morty" />
            <div className="item-prop">
        <span><b>Species:</b> {char.species}</span>
        <span><b>Gender:</b> {char.gender}</span>
        <span><b>Origin:</b> {char.origin.name}</span>
        <span><b>Status:</b> {char.status}</span>
        <span><b>Last Known Location:</b> {char.location.name}</span>
        </div>
        </div>
    );
}

export default CharacterCard;