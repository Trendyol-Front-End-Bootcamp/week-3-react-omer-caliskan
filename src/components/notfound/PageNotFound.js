import React from 'react';

function PageNotFound(){
    return(
        <div className="notfound">
        <img src="https://staticdelivery.nexusmods.com/mods/1151/images/528-0-1447526230.png" />
        <div>
         <h1>NO RESULT</h1>
        <h2>The page you are trying to search has been moved to another universe.</h2>
        </div>
        </div>
    )
}

export default PageNotFound;