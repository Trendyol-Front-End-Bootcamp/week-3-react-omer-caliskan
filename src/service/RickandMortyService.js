const axios = require('axios')

export const getCharacterById = async (charId) => {

  let character
  
  await axios
      .get(`https://rickandmortyapi.com/api/character/${charId}`)
      .then(({data}) => {character = data})

      return character;
}

export const getCharactersByPageandFilter = async (page, gender, status) => {

  let characters = []
  
  let query = `?page=${page}${gender!=="All" ? "&gender=" + gender : ""}${status!=="All" ? "&status=" + status : ""}`;

  await axios
      .get(`https://rickandmortyapi.com/api/character/${query}`)
      .then(({data}) => {characters = data})

  return characters;
}

export const getEpisodeName = async (episodesUrl) => {

  let episodeName;

  await axios
        .get(`${episodesUrl}`)
        .then(({data}) => {episodeName = data.name})
  
  return episodeName;
}