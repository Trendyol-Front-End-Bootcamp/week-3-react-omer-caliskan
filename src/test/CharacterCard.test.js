import { getCharacterById } from "../service/RickandMortyService";
import React from 'react'
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import CharacterCard from "../components/character/CharacterCard";
import PageNotFound from "../components/notfound/PageNotFound";


Enzyme.configure({ adapter: new Adapter() })

describe("Character Card", () => {
        let characterCard
        let char
        beforeEach(async () => {
          char = await getCharacterById(2)
          characterCard = shallow(<CharacterCard key={2} char={char} />)
        })
  
        it("should render character card", () => {
          expect(characterCard).toMatchSnapshot()
        })

        it("should character card have right class", () => {
            const checkClass = characterCard.hasClass("list-item")
            expect(checkClass).toBeTruthy()
          })

          it("should seen right character name of item", () => {
            const cardName = characterCard.find('h3').text()
            const charName = char.name

            expect(cardName).toEqual(charName)
          })

          describe("Not Found Page", () => {
            let notfound

            beforeEach(() => {
                notfound = shallow(<PageNotFound />)
            })

            it("should check page not found class", () => {
                const checkClass = notfound.hasClass("notfound")
        
                expect(checkClass).toBeTruthy()
              })

            it("should render Page Not Found", () => {
                expect(notfound).toMatchSnapshot()
            })

            it("should have right Page Not Found message", () => {
                const message = notfound.find('h2').text()

                expect(message).toEqual("The page you are trying to search has been moved to another universe.")
            })
          })
  
  
      })