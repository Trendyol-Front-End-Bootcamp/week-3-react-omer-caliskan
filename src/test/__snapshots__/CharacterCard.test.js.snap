// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Card should render character card 1`] = `
"<div className=\\"list-item\\">
  <h3>
    Morty Smith
  </h3>
  <img src=\\"https://rickandmortyapi.com/api/character/avatar/2.jpeg\\" alt=\\"Rick and Morty\\" />
  <div className=\\"item-prop\\">
    <span>
      <b>
        Species:
      </b>
       
      Human
    </span>
    <span>
      <b>
        Gender:
      </b>
       
      Male
    </span>
    <span>
      <b>
        Origin:
      </b>
       
      Earth (C-137)
    </span>
    <span>
      <b>
        Status:
      </b>
       
      Alive
    </span>
    <span>
      <b>
        Last Known Location:
      </b>
       
      Earth (Replacement Dimension)
    </span>
  </div>
</div>"
`;
